aws-xray/src/main/java/io/opentelemetry/contrib/awsxray/AwsSpanMetricsProcessor.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.contrib.awsxray;
+
+import static io.opentelemetry.semconv.trace.attributes.SemanticAttributes.HTTP_STATUS_CODE;
+
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.api.metrics.DoubleHistogram;
+import io.opentelemetry.api.metrics.LongCounter;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.sdk.resources.Resource;
+import io.opentelemetry.sdk.trace.ReadWriteSpan;
+import io.opentelemetry.sdk.trace.ReadableSpan;
+import io.opentelemetry.sdk.trace.SpanProcessor;
+import io.opentelemetry.sdk.trace.data.SpanData;
+import javax.annotation.concurrent.Immutable;
+
+/**
+ * This processor will generate metrics based on span data. It depends on a {@link
+ * MetricAttributeGenerator} being provided on instantiation, which will provide a means to
+ * determine attributes which should be used to create metrics. A {@link Resource} must also be
+ * provided, which is used to generate metrics. Finally, two {@link LongCounter}'s and a {@link
+ * DoubleHistogram} must be provided, which will be used to actually create desired metrics (see
+ * below)
+ *
+ * <p>AwsSpanMetricsProcessor produces metrics for errors (e.g. HTTP 4XX status codes), faults (e.g.
+ * HTTP 5XX status codes), and latency (in Milliseconds). Errors and faults are counted, while
+ * latency is measured with a histogram. Metrics are emitted with attributes derived from span
+ * attributes.
+ *
+ * <p>For highest fidelity metrics, this processor should be coupled with the {@link
+ * AlwaysRecordSampler}, which will result in 100% of spans being sent to the processor.
+ */
+@Immutable
+public final class AwsSpanMetricsProcessor implements SpanProcessor {
+
+  private static final double NANOS_TO_MILLIS = 1_000_000.0;
```

Comment:
nit: Shouldn't it be named `MILLIS_TO_NANOS` since 1 ms = 1000000 ns?

Reply:
It's used to convert nanos to millis, so prefer to leave it as it is. Is a multiplication/division factor, so either name is probably reasonable.